src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORM = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate fromDate = LocalDate.parse(dateFrom.trim(), DATE_FORM);
+        LocalDate toDate = LocalDate.parse(dateTo.trim(), DATE_FORM);
+
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+
+        for (String name : names) {
+            int totalEarned = 0;
+            for (String entry : data) {
+                String[] parts = entry.split(" ");
+                LocalDate entryDate = LocalDate.parse(parts[0], DATE_FORM);
+
+                if (name.equals(parts[1]) && isWithinDateRange(entryDate, fromDate, toDate)) {
+                    int hours = Integer.parseInt(parts[2]);
+                    int rate = Integer.parseInt(parts[3]);
+                    totalEarned += hours * rate;
+                }
+            }
+            report.append(name)
+                    .append(" - ")
+                    .append(totalEarned)
+                    .append(System.lineSeparator());
+        }
+
+        return report.toString().trim();
```

Comment:
The `trim` method call at the end of the `toString()` method is unnecessary because `System.lineSeparator()` does not add any extra spaces that need to be trimmed. You can return `report.toString()` directly.